apps/sms/services/test/unit/message_manager_test.js

```diff
@@ -412,91 +412,126 @@ suite('message_manager.js >', function() {
   });
 
   suite('markThreadRead()', function() {
-    setup(function() {
-      this.sinon.spy(MockNavigatormozMobileMessage, 'getMessages');
-      this.sinon.spy(MockNavigatormozMobileMessage, 'markMessageRead');
-
-      MessageManager.markThreadRead(1);
-    });
+    var cursor, threadId;
 
-    test('call mark read on the correct messages', function() {
-      assert.ok(
-        MockNavigatormozMobileMessage.getMessages.calledWithMatch({
-          threadId: 1,
-          read: false
-        })
-      );
+    setup(function() {
+      threadId = 1;
 
-      var messagesList = [
+      var testMessages = [
         { id: 1 },
         { id: 2 },
         { id: 3 },
         { id: 4 }
       ];
 
-      MockNavigatormozMobileMessage.mTriggerMessagesRequest(messagesList);
+      var filter = {
+        threadId: threadId,
+        read: false
+      };
 
-      MockNavigatormozMobileMessage.mTriggerMarkReadSuccess();
-      MockNavigatormozMobileMessage.mTriggerMarkReadSuccess();
-      MockNavigatormozMobileMessage.mTriggerMarkReadSuccess();
-      MockNavigatormozMobileMessage.mTriggerMarkReadSuccess();
+      cursor = {
+        done: false,
+        continue: function() {
+          var next = testMessages.shift();
+          if (next === undefined) {
+            this.result = null;
+            this.done = true;
+          } else {
+            this.result = next;
+          }
+          cursor.onsuccess();
+        }
+      };
 
-      // doing one more to see if we're not called too many times
-      MockNavigatormozMobileMessage.mTriggerMarkReadSuccess();
-      assert.equal(MockNavigatormozMobileMessage.markMessageRead.callCount, 4);
+    this.sinon.stub(MockNavigatormozMobileMessage, 'getMessages').withArgs(
+      filter, false).returns(cursor);
+    this.sinon.spy(MessageManager, 'markMessagesRead');
+  });
 
-      messagesList.forEach(function(message) {
-        assert.ok(
-          MockNavigatormozMobileMessage.markMessageRead.calledWith(
-            message.id, true
-          )
-        );
-      });
+    test('call mark read on the correct messages', function() {
+      MessageManager.markThreadRead(threadId);
+      cursor.continue();
+    
+      sinon.assert.calledWith(
+        MockNavigatormozMobileMessage.getMessages, {
+          threadId: 1,
+          read: false
+        }, false);
```

Comment:
nit: indentation in SMS is something like:
```javascript
sinon.assert.calledWith(
  MockNavigatormozMobileMessage.getMessages,
  {
    threadId: 1,
    read: false
  },
  false
);
```
(possibly you can put the object on one line)